python/ray/data/tests/test_dataset.py

```diff
@@ -1868,14 +1916,18 @@ def test_iter_batches_basic(ray_start_regular_shared):
 
     # Prefetch with ray.wait.
     context = DatasetContext.get_current()
-    context.actor_prefetcher_enabled = False
-    batches = list(
-        ds.iter_batches(prefetch_blocks=1, batch_size=None, batch_format="pandas")
-    )
-    assert len(batches) == len(dfs)
-    for batch, df in zip(batches, dfs):
-        assert isinstance(batch, pd.DataFrame)
-        assert batch.equals(df)
+    old_config = context.actor_prefetcher_enabled
+    try:
+        context.actor_prefetcher_enabled = False
+        batches = list(
+            ds.iter_batches(prefetch_blocks=1, batch_size=None, batch_format="pandas")
+        )
+        assert len(batches) == len(dfs)
+        for batch, df in zip(batches, dfs):
+            assert isinstance(batch, pd.DataFrame)
+            assert batch.equals(df)
+    finally:
+        context.actor_prefetcher_enabled = old_config
```

Comment:
Why do we need a try-catch here?

Reply:
To restore the old `actor_prefetcher_enabled` config value if the test fails, otherwise all of the below tests will be run with `context.actor_prefetcher_enabled=False`, which is not the default.